packages/map-template/src/hooks/useOutsideMapsIndoorsDataClick.js

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from 'react';
+import mapTypeState from '../atoms/mapTypeState';
+import { useRecoilValue } from 'recoil';
+import { mapTypes } from '../constants/mapTypes';
+
+/**
+ * Custom hook that handles map click events and determines whether the user clicked outside MapsIndoors data.
+ *
+ * @param {Object} mapsIndoorsInstance - The MapsIndoors instance.
+ * @param {boolean} isOpen - Whether the sidebar is open or not.
+ * @returns {boolean} Returns `true` if the user clicked outside MapsIndoors data, `false` otherwise.
+ */
+export default function useOutsideMapsIndoorsDataClick(mapsIndoorsInstance, isOpen) {
+    const mapType = useRecoilValue(mapTypeState);
+    const [clickedOutside, setClickedOutside] = useState(false);
+    const map = mapsIndoorsInstance.getMap();
+
+    // Reset clickedOutside state when the sidebar is opened
+    useEffect(() => {
+        if (isOpen) {
+            setClickedOutside(false);
+        }
+    }, [isOpen]);
+
+    useEffect(() => {
+        let mapboxClickHandler;
+        let googleMapsClickHandler;
+        let googleMapsIndoorsClickHandler;
+
+        if (mapType === mapTypes.Mapbox) {
```

Comment:
For better readability, try to extract the Mapbox and Google Maps specific logic into separate functions (meaning everything inside the if statements that look on mapType).